class CarException extends Exception {
	
}

class Car{
	private int num;
	private double gas;
	
	public Car() {
		num = 0;
		gas = 0.0;
		System.out.println("車を作成しました");
	}
	public void setCar(int n, double g) throws CarException{
		if(g < 0) {
			CarException e = new CarException();
			throw e;
		} else {
			num = 0;
			gas = 0;
			System.out.println("ナンバーを" + num + "にガソリン量を" + gas + "にしました");
		}
	}
	public void show() {
		System.out.println("車のナンバーは" + num + "です");
		System.out.println("ガソリン量は" + gas + "です");
	}
}
public class Sample14_5 {
	public static void main(String[] args) {
		Car car1 = new Car();
		try {
			car1.setCar(1234, 10.0);
		} catch(CarException e) {
			System.out.println(e + "が創出されました");
		}
		car1.show();
	}

}
